massa-bootstrap/src/messages.rs

```diff
@@ -123,6 +131,17 @@ impl DeserializeCompact for BootstrapMessage {
 
                 BootstrapMessage::FinalState { final_state }
             }
+            MessageTypeId::BootstrapError => {
+                let (error_len, delta) = u32::from_varint_bytes(&buffer[cursor..])?;
+                cursor += delta;
+
+                let error = String::from_utf8_lossy(&buffer[cursor..cursor + error_len as usize]);
```

Comment:
Critical attack vector: this will panic if error_len is longer than the buffer. Use buffer.get(cursor..cursor + xxxx) that returns None if the range is unavailable